src/Compilers/CSharp/Portable/BoundTree/BoundNodes.xml

```diff
@@ -1881,8 +1882,9 @@
   <Node Name="BoundCollectionLiteralExpression" Base="BoundExpression">
     <!-- Non-null type is required for this node kind -->
     <Field Name="Type" Type="TypeSymbol" Override="true" Null="disallow"/>
+    <Field Name="CollectionTypeKind" Type="CollectionLiteralTypeKind"/>
     <!-- An expression placeholder value representing the initialized collection. -->
-    <Field Name="Placeholder" Type="BoundObjectOrCollectionValuePlaceholder" Null="disallow" SkipInVisitor="true"/>
+    <Field Name="Placeholder" Type="BoundObjectOrCollectionValuePlaceholder?" Null="allow" SkipInVisitor="true"/>
```

Comment:
generally only null in error scenarios?

Reply:
Yes, this should only be `null` in error scenarios.